### DISCUSSION TITLE
Components structure

### DISCUSSION THREAD

**Discussion by zvitek** [nuxt/nuxt#8269] [Questions (Nuxt 2)]
Hello, I would like to know your opinions or practices about component structure in application.
I know, there is a folder `pages` for views/routing and folder `components`.

But I just missing something special or maybe don't understand these folders.

Don't let talk about folder `pages`, it's a folder with clear logic and it's ok.
But the folder `components`. In my understanding it's a folder for common components like `button`, `headlines` or some small parts of views.

What about rest of the parts?
The application have a lot of pages and views. Some pages are unique and are composed of unique parts for these page.
Where can I place these components?
- If into `components` folder. Which structure do you use?
- Nowhere else? Where?

Thank you for share your know how

**Comment by TheAlexLichter** [comment]
It's a matter of preference in the end.
My personal preference: I put **each component that's not a page** inside components and try to keep component names specific. Thus, I use a flat structure with almost no subfolders.

**Reply by zvitek** [reply]
In you case, you have to have much files/folders in`components` folder and long names of files, have you?

**Reply by TheAlexLichter** [reply]
Yes, depending on the App Size I have plenty of files in the `components` folder, sometimes with longer names.
But with Autocomplete that's not an issue IMO 😋 
Even better - it is easier to search for.
